### PR TITLE
CEDS-3579 Correct and simplify date validation

### PR DESCRIPTION
**Pull request by magicaljellybeans**
Add test case for all three fields being empty and add explanatory comments

### CODE REVIEW

File: app/forms/common/Date.scala [hunk 1]
```diff
@@ -55,58 +55,50 @@ object Date {
   val twoDigitFormatter = new DecimalFormat("00")
   val fourDigitFormatter = new DecimalFormat("0000")
 
-  def validate(day: Try[Int], month: Try[Int], year: Try[Int]): Boolean = build(day, month, year).isSuccess
-  def validateNonEmptyFields(day: String, month: String, year: String): Boolean = Seq(day, month, year).forall(_.nonEmpty)
-  def isAnyFieldNotEmpty(fields: Seq[String]): Condition = mapping => fields.exists(field => mapping.getOrElse(field, "").nonEmpty)
+  def isValidDate(day: String, month: String, year: String): Boolean =
```
Comment (glidester) [inline app/forms/common/Date.scala:58]: Maybe rename to `isValidDateOrEmpty

File: app/forms/common/Date.scala [hunk 2]
```diff
@@ -55,58 +55,50 @@ object Date {
...
+    if (isAnyFieldEmpty(Seq(day, month, year))) true
+    else Try(LocalDate.of(year.toInt, month.toInt, day.toInt)).isSuccess
+  def isAnyFieldNotEmpty(fields: Seq[String]): Boolean = fields.exists(_.nonEmpty)
```
Comment (glidester) [inline app/forms/common/Date.scala:61]: Maybe rename to `isAnyFieldPopulated` (save the sort of double negative use of 'not' & 'empty' together)?

File: app/forms/common/Date.scala [hunk 3]
```diff
@@ -55,58 +55,50 @@ object Date {
...
+  def isAnyFieldEmpty(fields: Seq[String]): Boolean = fields.exists(_.isEmpty)
+  def isAnyFieldNotEmptyCondition(fields: Seq[String]): Condition = mapping => fields.exists(field => mapping.getOrElse(field, "").nonEmpty)
 
   def mapping(prefix: String): Mapping[Date] =
     Forms
       .tuple(dayKey -> dayMapping(prefix), monthKey -> monthMapping(prefix), yearKey -> yearMapping(prefix))
```
Comment (glidester) [inline app/forms/common/Date.scala:67]: Maybe add a comment above this line stating "these checks only fire if at least one of these three fields contains some value, else these checks are skipped)"

Comment (magicaljellybeans) [reply app/forms/common/Date.scala:67]: That could be a bit misleading though because the individual field mappings do do some checking, even if it's just to see if the field is empty.

I've tried to explain what you've mentioned on lines 78-81 - don't know if you've seen that or I need to try and word it better?

**Review by glidester — APPROVED** [review]
